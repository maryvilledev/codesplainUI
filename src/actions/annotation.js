export const OPEN_ANNOTATION_PANEL = 'OPEN_ANNOTATION_PANEL';
export const CLOSE_ANNOTATION_PANEL = 'CLOSE_ANNOTATION_PANEL';

export const openAnnotationPanel = (snippetInformation) => ({
  type: OPEN_ANNOTATION_PANEL,
  payload: snippetInformation,
});

export const closeAnnotationPanel = () => ({
  type: CLOSE_ANNOTATION_PANEL,
});
