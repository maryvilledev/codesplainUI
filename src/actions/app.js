import axios from 'axios';

export const EDIT_ANNOTATION = 'EDIT_ANNOTATION';
export const PARSE_SNIPPET = 'PARSE_SNIPPET';
export const RESTORE_STATE = 'RESTORE_STATE';
export const SAVE_ANNOTATION = 'SAVE_ANNOTATION';
export const SET_AST = 'SET_AST';
export const SET_RULE_FILTERS = 'SET_RULE_FILTERS';
export const SET_SNIPPET_CONTENTS = 'SET_SNIPPET_CONTENTS';
export const SET_SNIPPET_LANGUAGE = 'SET_SNIPPET_LANGUAGE';
export const SET_SNIPPET_TITLE = 'SET_SNIPPET_TITLE';
export const TOGGLE_EDITING_STATE = 'TOGGLE_EDITING_STATE';
export const SAVE_STATE = 'SAVE_STATE';
export const SAVE_STATE_STARTED = 'SAVE_STATE_STARTED';
export const SAVE_STATE_SUCCEEDED = 'SAVE_STATE_SUCCEEDED';
export const SAVE_STATE_FAILED = 'SAVE_STATE_FAILED';


export const setSnippetContents = (snippet) => ({
  type: SET_SNIPPET_CONTENTS,
  payload: snippet,
});

export const setRuleFilters = (filters) => ({
  type: SET_RULE_FILTERS,
  payload: filters,
});

export const setAST = (AST) => ({
  type: SET_AST,
  payload: AST,
});

export const toggleEditState = () => ({
  type: TOGGLE_EDITING_STATE,
});

export const saveAnnotation = (annotationData) => ({
  type: SAVE_ANNOTATION,
  payload: annotationData,
});

export const restoreState = (savedState) => ({
  type: RESTORE_STATE,
  payload: savedState,
});

export const setSnippetTitle = (snippetTitle) => ({
  type: SET_SNIPPET_TITLE,
  payload: snippetTitle
})

export const setSnippetLanguage = (snippetLanguage) => ({
  type: SET_SNIPPET_LANGUAGE,
  payload: snippetLanguage
});

export const parseSnippet = (snippet) => ({
  type: PARSE_SNIPPET,
  meta: {
    WebWorker: true,
  },
  payload: {
    snippet,
  },
});

export const saveStateStarted = () => ({
  type: SAVE_STATE_STARTED,
});

export const saveStateSucceeded = () => ({
  type: SAVE_STATE_SUCCEEDED,
});

export const saveStateFailed = () => ({
  type: SAVE_STATE_FAILED,
});

export const saveState = (id) => {
  return (dispatch, getState) => {
    // If the snippet has not been saved, then saving an annotation shouldn't
    // trigger a POST request to save the snippet (and its annotations)
    if (id === undefined) {
      // Return a Promise object to make this action "thenable"
      return Promise.resolve();
    }
    dispatch(saveStateStarted());
    // Get the app state and stringify it
    const stateString = JSON.stringify(getState().app);

    // TODO: Dispatch action to show the snackbar with a relevant message based on the results
    // Return the results of the POST request
    return axios.post(`api/snippets/${id}`, { json: stateString, })
      .then((res) => {
        dispatch(saveStateSucceeded());
      }, (err) => {
        dispatch(saveStateFailed());
      });
  };
};
