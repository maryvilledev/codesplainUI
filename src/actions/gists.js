export const SET_GISTS = 'SET_GISTS';

export const setGists = gists => ({
  type: SET_GISTS,
  payload: gists,
});
