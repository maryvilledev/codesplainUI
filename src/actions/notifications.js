export const ADD_NOTIFICATION = 'ADD_NOTIFICATION';
export const CLOSE_NOTIFICATION = 'CLOSE_NOTIFICATION';

export const addNotification = notification => ({
  type: ADD_NOTIFICATION,
  payload: notification,
});

export const closeNotification = () => ({
  type: CLOSE_NOTIFICATION,
});
