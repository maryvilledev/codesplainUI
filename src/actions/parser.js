export const LOAD_PARSER = 'LOAD_PARSER'

export const loadParser = (parserURL) => ({
  type: LOAD_PARSER,
  meta: {
    WebWorker: true,
  },
  payload: parserURL,
})
