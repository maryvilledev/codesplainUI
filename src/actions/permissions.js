export const SET_PERMISSIONS = 'SET_PERMISSIONS';

export const setPermissions = permissions => ({
  type: SET_PERMISSIONS,
  payload: permissions,
});
