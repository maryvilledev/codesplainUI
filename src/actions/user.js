import axios from 'axios';
import cookie from 'react-cookie';

import { makeSaveEndpointUrl } from '../util/requests';

const API_URL = process.env.REACT_APP_API_URL;

// Util func to check for 'NoSuchKey' responses from S3
const noSuchKey = data => (
  typeof data === 'string' && data.includes('NoSuchKey')
);

export const SET_USER_SNIPPETS = 'SET_USER_SNIPPETS';
export const UPDATE_USER_SNIPPETS_STARTED = 'UPDATE_USER_SNIPPETS_STARTED';
export const UPDATE_USER_SNIPPETS_SUCCEEDED = 'UPDATE_USER_SNIPPETS_SUCCEEDED';
export const UPDATE_USER_SNIPPETS_FAILED = 'UPDATE_USER_SNIPPETS_FAILED';
export const UPDATE_USER_SNIPPETS = 'UPDATE_USER_SNIPPETS';
export const SAVE_USERNAME = 'SAVE_USERNAME';
export const SAVE_ACCESS_TOKEN = 'SAVE_ACCESS_TOKEN';
export const CLEAR_USER_CREDENTIALS = 'CLEAR_USER_CREDENTIALS';
export const RESTORE_USER_CREDENTIALS = 'RESTORE_USER_CREDENTIALS';

export const saveUsername = username => ({
  type: SAVE_USERNAME,
  payload: username,
});

export const saveAccessToken = accessToken => ({
  type: SAVE_ACCESS_TOKEN,
  payload: accessToken,
});

export const clearUserCredentials = () => ({
  type: CLEAR_USER_CREDENTIALS,
});

export const setUserSnippets = snippetMeta => ({
  type: SET_USER_SNIPPETS,
  payload: snippetMeta,
});

export const updateUserSnippetsStarted = () => ({
  type: UPDATE_USER_SNIPPETS_STARTED,
});

export const updateUserSnippetsSucceeded = () => ({
  type: UPDATE_USER_SNIPPETS_SUCCEEDED,
});

export const updateUserSnippetsFailed = () => ({
  type: UPDATE_USER_SNIPPETS_FAILED,
});

export const updateUserSnippets = () => (dispatch) => {
  // Load requisite cookies, return Promise if they aren't present
  const token = cookie.load('token');
  const username = cookie.load('username');

  // Fetch the user's snippet meta data and save it
  const headers = {
    Accept: 'application/json',
    Authorization: `token ${token}`,
  };
  dispatch(updateUserSnippetsStarted());
  return axios.get(makeSaveEndpointUrl(username), { headers })
    .then((res) => {
      // Jump to catch block if the user has no index.json file:
      if (noSuchKey(res.data)) {
        throw new Error(`index.json does not exist for ${username}!`);
      }
      dispatch(setUserSnippets(res.data));
      dispatch(updateUserSnippetsSucceeded());
    })
    .catch(() => {
      dispatch(updateUserSnippetsFailed());
    });
};

export const fetchAccessToken = authCode => () => {
  const reqUrl = `${API_URL}/auth`;
  const reqBody = { code: authCode };
  return axios({
    method: 'POST',
    url: reqUrl,
    data: reqBody,
  });
};

export const fetchUserInfo = () => (dispatch, getState) => {
  const { token } = getState().user;
  const reqHeaders = {
    Accept: 'application/json',
    Authorization: `token ${token}`,
  };
  return axios({
    method: 'GET',
    url: 'https://api.github.com/user',
    headers: reqHeaders,
  });
};

export const restoreUserCredentials = (token, username) => ({
  type: RESTORE_USER_CREDENTIALS,
  payload: {
    token,
    username,
  },
});
