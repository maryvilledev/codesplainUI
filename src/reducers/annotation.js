import * as actions from '../actions/annotation';

export const initialState = {
  isDisplayingAnnotation: false,
  snippetInformation: {},
};

const annotation = (state = initialState, action) => {
  switch (action.type) {
    case actions.OPEN_ANNOTATION_PANEL: {
      return {
        ...state,
        isDisplayingAnnotation: true,
        snippetInformation: action.payload,
      };
    }
    case actions.CLOSE_ANNOTATION_PANEL: {
      if (!state.isDisplayingAnnotation) {
        return state;
      }
      return {
        ...state,
        isDisplayingAnnotation: false,
        snippetInformation: {},
      };
    }
    default: {
      return state;
    }
  }
}

export default annotation;
