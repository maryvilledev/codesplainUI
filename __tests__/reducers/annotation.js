import * as actions from '../../src/actions/annotation';
import reducer, { initialState } from '../../src/reducers/annotation';

describe('Reducer: Annotation', () => {
  describe('initialState', () => {
    it('should match snapshot', () => {
      expect(reducer(undefined, {})).toMatchSnapshot();
    });
  });

  it('should handle OPEN_ANNOTATION_PANEL', () => {
    const lineAnnotated = {
      lineNumber: 1,
      lineText: 'Wubba lubba dub dub!',
    };
    const action = {
      type: actions.OPEN_ANNOTATION_PANEL,
      payload: lineAnnotated,
    };
    const expected = {
      ...initialState,
      isDisplayingAnnotation: true,
      lineAnnotated: action.payload,
    };
    expect(reducer(undefined, action)).toEqual(expected);
  });

  it('should handle CLOSE_ANNOTATION_PANEL', () => {
    const action = {
      type: actions.CLOSE_ANNOTATION_PANEL,
    };
    const expected = {
      ...initialState,
      isDisplayingAnnotation: false,
      lineAnnotated: {},
    };
    expect(reducer(undefined, action)).toEqual(expected);
  });
});
