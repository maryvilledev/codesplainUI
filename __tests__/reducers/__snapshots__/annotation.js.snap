exports[`Reducer: Annotation initialState should match snapshot 1`] = `
Object {
  "isDisplayingAnnotation": false,
  "lineAnnotated": Object {},
}
`;
